### PR TITLE
Adding speech recognition and synthesis for web chat

### PR DESCRIPTION
**Pull request by smacdo**

### CODE REVIEW

File: src/AudioRecorder.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    private context: any;
+    private node: any;
+    private audioActive: boolean;
+    private recording: boolean;
+    private websocket: WebSocket;
+    private onAudioFlowing: Action;
+    private readonly bufferLen = 4096;
+
+
+    //todo: add on audio active
```
Comment (GeekTrainer) [inline src/AudioRecorder.ts]: Has this been completed? We don't want the comment in the merged code.

File: src/AudioRecorder.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            this.audioActive = true;
+            if (this.onAudioFlowing) {
+                this.onAudioFlowing();
+                this.onAudioFlowing = null;
+            }
+
+            if (!this.recording)
+                return;
+
+            // Todo, probably buffer here while websocket connects.
```
Comment (GeekTrainer) [inline src/AudioRecorder.ts]: Todo comment...

**Comment by danmarshall** [comment]
Do we want this to go to the master branch?

**Comment by GeekTrainer** [comment]
Going to close this

**Comment by danmarshall** [comment]
Closing this

**Pull request closed without merging**